node/src/components/block_synchronizer/block_builder/tests.rs

```diff
@@ -0,0 +1,198 @@
+use std::collections::BTreeMap;
+
+use casper_types::testing::TestRng;
+use num_rational::Ratio;
+
+use crate::components::consensus::tests::utils::{ALICE_PUBLIC_KEY, ALICE_SECRET_KEY};
+
+use super::*;
+
+#[test]
+fn handle_acceptance() {
+    let mut rng = TestRng::new();
+    let block = Block::random(&mut rng);
+    let mut builder = BlockBuilder::new(
+        block.header().block_hash(),
+        false,
+        1,
+        TimeDiff::from_seconds(1),
+        TimeDiff::from_seconds(1),
+    );
+
+    let honest_peer = NodeId::random(&mut rng);
+    let dishonest_peer = NodeId::random(&mut rng);
+
```

Comment:
Please add some comments for each scenario group tested here.
```suggestion
// Test acceptance for data that was not received from peers.
```

Reply:
Done.